airbyte-integrations/connectors/source-s3/source_s3/stream.py

```diff
@@ -53,10 +53,10 @@ def _list_bucket(self, accept_key: Callable = lambda k: True) -> Iterator[FileIn
             except KeyError:
                 pass
             else:
-                for c in content:
-                    key = c["Key"]
-                    if accept_key(key):
-                        yield FileInfo(key=key, last_modified=c["LastModified"], size=c["Size"])
+                for file in content:
```

Comment:
As I remember we agreed move this functionality to `need_to_skip_file` method

Reply:
yes, but I've decided to leave it here, as `need_to_skip_file` used only for incremental sync, while we need this functionality for full refresh as well.

Reply:
Let's refactor this logic and add filtrating depend on sync type. 

 PCODE:
```
date = (stream_state or {}).get(cursor_field, start_date):
if last_mod > date: 
   process file 
```

Reply:
it seems to be not the optimal solution. 
Method `need_to_skip_file` is used only in incremental sync, and works with `stream_state` and `history` objects.

My goal is to modify the `list_objects` S3 method in such a way that the connector will basically only works with the filtered files.

Reply:
updated